### PR TITLE
Skip confirmation for `operator remove` if specifying `--dry-run`

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [x] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by zirain — APPROVED** [review]

**Comment by zirain** [comment]
worth a release-note for backport?

**Review by morvencao — APPROVED** [review]

**Review by zirain — APPROVED** [review]

File: releasenotes/notes/43120.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: feature
+area: installation
```
Comment (ericvn) [inline releasenotes/notes/43120.yaml:3]: Might this be better in the istioctl section, at least since it's --dry-run.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**